drivers/audio/bcm2836/common.h

```diff
@@ -346,5 +346,18 @@ NewAdapterCommon
     _In_        POOL_TYPE               PoolType 
 );
 
+void* __cdecl operator new
+(
+    size_t          iSize,
+    _When_((poolType& NonPagedPoolMustSucceed) != 0,
+        __drv_reportError("Must succeed pool allocations are forbidden. "
+            "Allocation failures cause a system crash"))
+    POOL_TYPE       poolType,
+    ULONG           tag
+);
 
-
+void __cdecl operator delete
+(
+    _Pre_maybenull_ __drv_freesMem(Mem) void* pVoid,
+    _In_ size_t cbSize
+);
```

Comment:
It's better/cleaner to have the new/delete operators in their own file, and include the header in `rpiwav.h`.
See: https://github.com/raspberrypi/windows-drivers/pull/3/commits/56501684b1959fdd2b89dd96be6074cae2845163 (it also includes the array delete methods)

This is how Microsoft addressed the issue in their sysvad driver sample (https://github.com/microsoft/Windows-driver-samples/tree/master/audio/sysvad), and

Reply:
I'm not sure I see any value in defining delete operators that are not used by the component.

I'm not sure I see any value in having a separate header for 2 definitions that should be included everywhere.

If you feel strongly about this, I will change it. I just don't see any benefit to having these in a separate cpp file or a separate header.

Reply:
I agree. The other delete operators can be added later on if ever needed.